sdk/eventhub/azure-eventhub/azure/eventhub/_tracing.py

```diff
@@ -0,0 +1,270 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+from __future__ import annotations
+from contextlib import contextmanager
+from enum import Enum
+import logging
+from typing import (
+    TYPE_CHECKING,
+    Type,
+    Optional,
+    Dict,
+    List,
+    Union,
+    Any,
+    Iterable,
+    Iterator
+)
+
+from azure.core import CaseInsensitiveEnumMeta
+from azure.core.settings import settings
+from azure.core.tracing import SpanKind, Link
+
+from .amqp import AmqpAnnotatedMessage
+from ._constants import PROP_TIMESTAMP
+
+if TYPE_CHECKING:
+    # pylint: disable=ungrouped-imports
+    from ._transport._base import AmqpTransport
+    try:
+        from uamqp import Message as uamqp_Message
+    except ImportError:
+        uamqp_Message = None
+    from ._pyamqp.message import Message
+    from azure.core.tracing import AbstractSpan
+    from ._client_base import ClientBase
+    from ._common import EventData, EventDataBatch
+
+
+_LOGGER = logging.getLogger(__name__)
+
+
+TRACE_DIAGNOSTIC_ID_PROPERTY = b"Diagnostic-Id"
+TRACE_PARENT_PROPERTY = b"traceparent"
+TRACE_STATE_PROPERTY = b"tracestate"
+TRACE_PROPERTY_ENCODING = "ascii"
+
+
+class TraceAttributes:
+    TRACE_NAMESPACE_ATTRIBUTE = "az.namespace"
+    TRACE_NAMESPACE = "Microsoft.EventHub"
+
+    TRACE_MESSAGING_SYSTEM_ATTRIBUTE = "messaging.system"
+    TRACE_MESSAGING_SYSTEM = "eventhubs"
+
+    TRACE_NET_PEER_NAME_ATTRIBUTE = "net.peer.name"
+    TRACE_MESSAGING_DESTINATION_ATTRIBUTE = "messaging.destination.name"
+    TRACE_MESSAGING_OPERATION_ATTRIBUTE = "messaging.operation"
+    TRACE_MESSAGING_BATCH_COUNT_ATTRIBUTE = "messaging.batch.message_count"
+
+    LEGACY_TRACE_COMPONENT_ATTRIBUTE = "component"
```

Comment:
It's totally up to you, but I believe this one can be removed completely - it was never used by anyone and is not really documented.

Reply:
Sure, I can remove this to clean the attributes up a bit.